code/script.js

```diff
@@ -301,25 +415,86 @@ const filterCharacters = (keep) => {
   */
 
   // Invoke a function to redraw the board with the remaining people.
-}
+ 
+
+ const filteredCharacters = CHARACTERS.filter( (singleCharacter) => {
+ if (singleCharacter.hair === 'black') {
+   return singleCharacter;
+ }  
+});
+
+filteredCharacters.forEach((charachter)=>  {
+charachter.hair = 'yellow'  
+})
+
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
-const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
-}
 
+
+const guess = (personToConfirm) => {      // store the interaction from the player in a variable.
+
+  let confirmation = confirm (`Are you sure that ${personToConfirm} is the secret person? `) // remember the confirm() ?
+  if (confirmation === true)
+{
+  checkMyGuess (personToConfirm)
+}   
+}
+// If the player wants to guess, invoke the checkMyGuess function.
 // If you confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+
+  if (personToCheck === secret.name) {    // 1. Check if the personToCheck is the same as the secret person's name
+    winOrLoseText.innerHTML (`You are a WINNER!`)
+ } else {
+    winOrLoseText.innerHTML (`I'm sorry! The secret person is ${secret.name}`)// 2. Set a Message to show in the win or lose section accordingly
+ }
+  
+  
+  winOrLose.style.display = 'flex' // 3. Show the win or lose section
+  board.style.display = 'none' // 4. Hide the game board
 }
 
+
+
+
+
+
+
+// How many questions did you need until you guessed right? Click up. 
+// Save yore score. Play again. Click up or down to register yor new score.  
+
+
+
+//  let saveEl = document.getElementById("save-el")
+//  let countEl = document.getElementById("count-el")
+
+
+
+ let count=0
+ function increment()
+ {
+ count+=1
+ countEl.innerText = count
+ }
+
+ function decrement()
+ {
+     count=count-1
+     countEl.innerText=count
+ }
+
+ function save()
+ {
+ let countStr = count + " - "
+     saveEl.innerText += countStr
+ }
+
+
 // Invokes the start function when website is loaded
-start()
+ start()
 
 // All the event listeners
 restartButton.addEventListener('click', start)
+ questions.addEventListener('change', selectQuestion)
+ board.addEventListener('click')
```

Comment:
The addEventListener needs two parameters, the second parameter should be the function you want to invoke.